### DISCUSSION TITLE
Is there any plan to support `keras-nlp` on macOs?

### DISCUSSION THREAD

**Discussion by alvarobartt** [keras-team/keras-hub#1032] [Q&A] [answered]
Hi everyone!

I tried to reproduce the example in the `README.md`, but in macOS it doesn't seem feasible, as `tensorflow_text` is not supported for ARM and it's required by all the tokenizers here, so one is not able to actually run e.g. `keras_nlp.models.BertClassifier.from_preset(...)` in macOS.

Is there any plan to support it? Not talking about MPS chips, but just in general.

Thanks in advance! 🙌🏻

**Comment by jbischof** [comment, marked as answer]
Hi @alvarobartt unfortunately this is a `tensorflow-text` issue and not something we control. However, adventurous users can build the library from source ([example guide](https://github.com/sun1638650145/Libraries-and-Extensions-for-TensorFlow-for-Apple-Silicon/blob/main/tutorials/text/text.md)). Let us know if this works for you!

The short story is that almost all machine learning takes places on linux servers and so windows and mac support never get high priority (at Google or elsewhere).

**Reply by alvarobartt** [reply]
Hi @jbischof thanks for the fast and detailed response! I'll try to look for a workaround to get it working on macOS, but for the moment Linux is OK for me, thanks!

**Reply by jbischof** [reply]
If you just want to demo our library without a lot of setup, I recommend colab.

**Reply by alvarobartt** [reply]
I was using Colab for the moment! BTW nice work with KerasNLP 🚀
